modules/calib3d/src/calibinit.cpp

```diff
@@ -471,6 +491,119 @@ static void icvBinarizationHistogramBased(Mat & img)
     }
 }
 
+ChessBoardDetector::NeighborsFinder::NeighborsFinder(ChessBoardDetector& _detector): detector(_detector) {
+    const int all_corners_count = detector.all_quads_count * 4;
+    neighbors_indices.resize(all_corners_count);
+    neighbors_dists.resize(all_corners_count);
+    all_quads_pts.reserve(all_corners_count);
+    for (int idx = 0; idx < detector.all_quads_count; idx++)
+    {
+        const ChessBoardQuad& cur_quad = (const ChessBoardQuad&)detector.all_quads[idx];
+        for (int i = 0; i < 4; i++)
+            all_quads_pts.push_back(cur_quad.corners[i]->pt);
+    }
+    const cvflann::KDTreeSingleIndexParams index_params;
+    p_all_quads_pts_index = makePtr<flann::GenericIndex<flann::L2_Simple<float>>>
```

Comment:
You can get rid of smart pointer, convert  `p_all_quads_pts_index` to regular field and use initializer list for it.